httplog/logger.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2022 The forwarder Authors. All rights reserved.
+// Use of this source code is governed by a MPL
+// license that can be found in the LICENSE file.
+
+package httplog
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/google/martian/v3/log"
+	"github.com/google/martian/v3/messageview"
+	"github.com/saucelabs/forwarder/middleware"
+)
+
+type LoggerMode string
+
+const (
+	URLLogMode     LoggerMode = "url"
+	HeadersLogMode LoggerMode = "headers"
+	BodyLogMode    LoggerMode = "body"
+	ErrOnlyLogMode LoggerMode = "error"
+)
+
+func (m LoggerMode) String() string {
+	return string(m)
+}
+
+// Logger is a modifier that logs requests and responses.
+type Logger struct {
+	log func(line string)
+}
+
+// NewLogger returns a logger that logs requests and responses, optionally
+// logging the body. Log function defaults to martian.Infof.
+func NewLogger() *Logger {
+	return &Logger{
+		log: func(line string) {
+			log.Infof(line)
+		},
+	}
+}
+
+func (l *Logger) Parse(val string) (middleware.Logger, error) {
```

Comment:
There is a problem with configuration you want to do too much at once. A logger needs a log sink and mode.
This should be read from the config and only then you can properly initialize it...
SetLogFunc should be removed.

Defaulting to martian log is wrong, for instance now HttpServer uses matianlog...